camel/agents/task_agent.py

```diff
@@ -151,3 +154,75 @@ def step(
 
         sub_tasks_msg = sub_tasks_msgs[0]
         return TextPrompt(sub_tasks_msg.content)
+
+
+class RoleAssignmentAgent(ChatAgent):
+    r"""
+    An agent that generates role names based on the task prompt.
+    Attributes:
+        role_assignment_prompt (TextPrompt): A prompt for the agent to generate
+        role names.
+    args:
+        model (ModelType): The tupe of model to use for the agent.
+            (default: :obj: 'ModelType.GPT_3_5_TURBO')
+        model_config (Any): The configuration for the model.
+            (default: :obj: 'None')
+    """
+
+    def __init__(
+        self,
+        model: ModelType = ModelType.GPT_3_5_TURBO,
+        model_config: Any = None,
+    ) -> None:
+        self.role_assignment_prompt = TextPrompt(
+            "Given this task, \"{task}\", generate two role names, one for the"
+            + " AI user and one for the AI assistant.")
+
+        system_message = SystemMessage(
+            role_name="Role Assigner",
+            role_type=RoleType.ASSISTANT,
+            content="You assign roles based on tasks.",
+        )
+
+        super().__init__(system_message, model, model_config)
+
+    def step(
+        self,
+        task_prompt: Union[str, TextPrompt],
+    ) -> Tuple[TextPrompt, TextPrompt]:
+        r""""
+        Generate role names based on the input task prompt.
+        Args:
+            task_prompt (Union[str, TextPrompt]): The prompt
+            for the task based
+            on which the roles are to be generated.
+            Returns:
+                Tuple[TextPrompt, Textprompt]: The role names generated by the
+                agent for
+                the AI user and AI assistant, respectively.
+        """
+
+        self.reset()
+
+        task_msg = AssistantChatMessage(
+            role_name="Role Assigner",
+            content=self.role_assignment_prompt.format(task=task_prompt))
+
+        role_msgs, terminated, _ = super().step(task_msg)
+
+        ai_user_role_name = None
+        ai_assistant_role_name = None
+
+        for message in role_msgs:
+            content = message.content
+            names = content.split("\n")
+
+            ai_user_role_name = names[0].split(":")[1].strip()
+            ai_assistant_role_name = names[1].split(":")[1].strip()
+
+        if ai_user_role_name is None or ai_assistant_role_name is None:
+            raise RuntimeError("Got None or insufficient Role messages. ")
+        if terminated:
+            raise RuntimeError("Role assignment failed.")
+
+        return ai_user_role_name, ai_assistant_role_name
```

Comment:
Let's make it a dictionary `dict(ai_user_role_name =..., ai_assistant_role_name=...)`